lib/rubocop/cop/rspec/factory_bot/attribute_defined_statically.rb

```diff
@@ -31,12 +31,15 @@ class AttributeDefinedStatically < Cop
             (send _ !#reserved_method? $...)
           PATTERN
 
-          def_node_search :factory_attributes, <<-PATTERN
+          def_node_matcher :factory_attributes, <<-PATTERN
             (block (send _ #attribute_defining_method? ...) _ { (begin $...) $(send ...) } )
           PATTERN
 
           def on_block(node)
-            factory_attributes(node).to_a.flatten.each do |attribute|
+            attributes = factory_attributes(node) || []
+            attributes = [attributes] if !attributes.is_a?(Array)
```

Comment:
TBH I haven't found a better way to handle all the result types. So any advice is welcome.

`factory_attributes` may return:
- array of nodes (when there are several attribute declarations - it matches `(begin $...)`)
- one node (when there is only one attribute - it matches `$(send ...)`)
- nil (when a node doesn't match the pattern)

Method generated with `def_node_search` returned an array even if there was only one attribute

Reply:
```
Array(1)
=> [1]
irb(main):005:0> Array([1])
=> [1]
irb(main):006:0> Array([1,2])
=> [1, 2]
irb(main):007:0> Array(nil)
=> []
```
so you can just do
```
attributes = Array(factory_attributes(node))
```

Reply:
Unfortunately, a `send` node deconstructs when wrapped with `Array()` and produces nonsense:
```ruby
> factory_attributes(node)
=> s(:send, nil, :author,
  s(:hash,
    s(:pair,
      s(:sym, :age),
      s(:int, 42)),
    s(:pair,
      s(:sym, :factory),
      s(:sym, :user))))

> Array(factory_attributes(node))
=> [nil, :author, s(:hash,
  s(:pair,
    s(:sym, :age),
    s(:int, 42)),
  s(:pair,
    s(:sym, :factory),
    s(:sym, :user)))]

> attributes
=> [s(:send, nil, :author,
  s(:hash,
    s(:pair,
      s(:sym, :age),
      s(:int, 42)),
    s(:pair,
      s(:sym, :factory),
      s(:sym, :user))))]
```

Reply:
Then you can use [attribites].flatten.compact, but I would preserve your version actually, as it's obvious what is going there